source/_posts/2019-01-14-use-or-not-to-use-++-operator-in-elixir.markdown

```diff
@@ -0,0 +1,260 @@
+---
+layout: post
+title: "Use or not to use ++ operator in Elixir"
+category: elixir
+comments: true
+author:
+  name: Jorge Bejar
+  email: jorge@wyeworks.com
+  twitter_handle: jmbejar
+  github_handle: jmbejar
+  image: /images/team/jorge-bejar.jpg
+  description: Love to explore and write about Elixir, Ruby, Javascript, full-stack development and technology in general.
+---
+
+If you're familiar with *Elixir* you may have been warned about the risks of using `++` operator to concatenate lists. In fact, a piece of advice is found directly on the [official documentation for this method](https://hexdocs.pm/elixir/Kernel.html#++/2):
+
+> The complexity of `a ++ b` is proportional to `length(a)`, so avoid repeatedly appending to lists of arbitrary length, e.g. `list ++ [item]`. Instead, consider prepending via `[item | rest]` and then reversing.
+
+In the rest of this article we are going to discuss an example and try to explain what could be problematic using `++` but also checking if we can make an improvement refactoring as suggested on the documentation. Spoiler: life is not so easy 😅
+
+<!--more-->
+
+## The problem
+
+Let's say we need to implement a function that receives a list of integers and returns a list including those numbers repeated several times, as it is showed in the following examples:
+
+`my_func([1]) # it returns [1]`
+`my_func([1,2]) # it returns [1,2,2]`
+`my_func([1,3,2]) # it returns [1,3,3,3,2,2]`
+
+The idea is to generate a list including each number *n times* where *n* is the actual value. We also expect that the original ordering of these numbers is preserved in the resulting list.
+
+## Our first naive implementation
+
+We could say that a very simple solution to this problem could be an approach where we use `Enum.reduce/3` and repeat each number, concatenating the list of repeated values to the accumulated list.
```

Comment:
This sentence feels a little repetitive to me because of the many `could`.

Maybe something like this might sound better (not really sure n_n")
```
A very simple solution to this problem could be using `Enum.reduce/3` and repeat each number ...
```

Reply:
Liked it